crates/configuration/src/network.rs

```diff
@@ -42,48 +60,114 @@ pub struct NetworkConfig {
 
     /// HRMP channels configurations.
     hrmp_channels: Vec<HrmpChannelConfig>,
+}
+
+/// A network configuration, composed of a relaychain, parachains and HRMP channels.
+#[derive(Debug, Default)]
+pub struct NetworkConfigBuilder<R> {
+    /// The global settings applied to the network.
+    global_settings: GlobalSettings,
+
+    /// Relaychain configuration.
+    relaychain: RelaychainConfig,
+
+    /// Parachains configurations.
+    parachains: Vec<ParachainConfig>,
+
+    /// HRMP channels configurations.
+    hrmp_channels: Vec<HrmpChannelConfig>,
 
-    // [TODO]: what does it represents ?
-    seed: String,
+    typestate: PhantomData<R>,
 }
 
-impl Default for NetworkConfig {
-    fn default() -> Self {
-        // [TODO]: define the default value for a network
-        todo!()
+impl NetworkConfigBuilder<NoRelayChain> {
+    pub fn new() -> Self {
+        NetworkConfigBuilder::default()
     }
 }
 
-impl NetworkConfig {
-    pub fn new() -> NetworkConfig {
-        Self::default()
+impl NetworkConfigBuilder<WithRelayChain> {
+    pub fn build(self) -> Result<NetworkConfig, ConfigError> {
```

Comment:
I think we should returns a Vec<ConfigError> with specific path of the failing key to guide the user.

For example having something like this:

Error: Cause of the error in "network.relaychain.default_command"

wdyt ?

Reply:
Sounds good, a Vec of error is great and we can keep in mind to implement in a follow-up pr.